src/machine/memory/frontend_memory.cpp

```diff
@@ -68,9 +68,10 @@ void FrontendMemory::write_ctl(enum AccessControl ctl, Address offset, RegisterV
             QString::number(ctl));
     }
     }
+    handle_control_signal(ctrl_info);
```

Comment:
Should not be required there. `ctrl_info` has to be propagated into functions to realize access. I would suggest to add parameter `ctrl_info` or better `AccesMode mode` before `value` to keep order `addeess`, `mode` consistent for read and write.